index.js

```diff
@@ -48,11 +50,52 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // split file array into chunks of 30
+      var i,j, chunks = [], chunkSize = 30;
+      for (i=0,j=files.length; i<j; i+=chunkSize) {
+          chunks.push( files.slice(i,i+chunkSize));
+      }
+
+      // launch a new process for each chunk
+      async.series(
+        chunks.map(function(chunk) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
+                                          args.concat(chunk),
+                                          {stdio: stdio});
+            clangFormatProcess.on('close', function(exit) {
+              if (exit !== 0) callback(exit);
+              else callback(null, exit);
+            });
+          };
+        }),
+        function(err, results) {
+          if (err) done(err);
+          console.log('\n');
+          console.log('ran clang-format with',
```

Comment:
I think the user would rather see 
`formatted 60 files`
instead of
`formatted 30 files\nformatted 30 files`

Also is there a way to print instead
```
re-formatting thing1.ts
re-formatting gulpfile.js
re-formatting README.md
re-formatting thing2.js
Checked formatting in 60 files. 4 files reformatted.
```
?